docs/Antora/modules/howto_guides/pages/howto_configure_instance_providers.adoc

```diff
@@ -0,0 +1,206 @@
+[tabs]
+====
+Bash::
++
+[source,bash]
+[]
+----
+export OPENDAQ_CONFIG_MODULEMANAGER_MODULESPATH="/tmp/modules/"
+export OPENDAQ_CONFIG_LOGGING_GlobalLogLevel=1
+export OPENDAQ_CONFIG_Modules_SomeModule_EnableSomething=true
+export OPENDAQ_CONFIG_Modules_SomeModule_FloatValue=12.5
+----
+====
+The `OPENDAQ_CONFIG_` prefix is case-sensitive and indicates that the variable is part of the instance builder options. The rest of the environment key is not case-sensitive and is considered as nodes of a dictionary split by "_". For example, `OPENDAQ_CONFIG_MODULEMANAGER_MODULESPATH="/tmp/modules/"` is represented as populating the instance builder options with the corresponding values. If a value was already set, it will be overridden with the new one in case the type of the new value matches the old one.
+[source,json]
+----
+{
+  "modulemanager": {
+    "modulespath": "/tmp/modules/"
+  }
+}
+----
+If value was already set, it will be override with new one in case if type of new value is the same as for old one.
+
+Example of creating Instance from enviorment variables
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+
+int main(int argc, const char* argv[])
+{
+    const ConfigProviderPtr configProvider = EnvConfigProvider();
+    const InstanceBuilderPtr instanceBuilder = InstanceBuilder().addConfigProvider(configProvider);
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder();
+
+    return 0;
+}
+----
+Python::
++
+[source,python]
+----
+import opendaq
+
+config_provider = opendaq.EnvConfigProvider()
+instance_builder = opendaq.InstanceBuilder()
+instance_builder.add_config_provider(config_provider)
+instance = instance_builder.build()
+----
+====
+
+== Instance Configuration with Command Line Arguments
+
+An instance can be configured from command line arguments. If `CmdLineArgsConfigProvider` is set to the builder, the provider will parse all command line arguments starting with `--config-`. The structure of the command line arguments reflects the hierarchy of the configuration, with each level separated by a dash (`-`). For example, `--config-modulemanager-modulespath="testtest"` would be recognized and parsed accordingly. Note that values are treated similarly to environment variables: if enclosed in quotes, they are recognized as string types; otherwise, the provider attempts to parse them as integer, float, and boolean types. If parsing fails, the value is considered as a string type.
+
+[tabs]
+====
+Bash::
++
+[source,bash]
+[]
+----
+./program --config-modulemanager-modulespath="testtest" --config-LOGGING-GlobalLogLevel=1
+--config-Modules-SomeModule-EnableSomething=true --config-Modules-SomeModule-FloatValue=12.5
```

Comment:
I think we should use underscores as delimiter here as well. To shorten the above, we could do it in a manner similar to CMake:

`./program -Cmodulemanager_modulespath="testtest"`

Also - let's avoid values such as "testtest" in documentation, and add a more representative path :)